src/aatoolbox/datasources/glofas/glofas.py

```diff
@@ -0,0 +1,456 @@
+"""Base class for downloading and processing GloFAS river discharge data."""
+import logging
+import time
+from abc import abstractmethod
+from dataclasses import dataclass
+from datetime import date
+from pathlib import Path
+from typing import List, Union
+
+import cdsapi
+import numpy as np
+import xarray as xr
+from dateutil import rrule
+
+from aatoolbox.config.countryconfig import CountryConfig
+from aatoolbox.datasources.datasource import DataSource
+from aatoolbox.utils.geoboundingbox import GeoBoundingBox
+
+_MODULE_BASENAME = "glofas"
+_HYDROLOGICAL_MODEL = "lisflood"
+_RIVER_DISCHARGE_VAR = "dis24"
+_REQUEST_SLEEP_TIME = 60  # seconds
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass
+class _QueryParams:
+    """
+    Class to keep track of CDS query input and output.
+
+    Parameters
+    ----------
+    filepath: Path
+        Full filepath of downloaded CDS file
+    query: dict
+        Output of _get_query() method, to be submitted to the CDS API
+    request_id: str, default = None
+        Request ID from CDS, only set after request has been made
+    downloaded: bool, default = False
+        Whether or not the file has yet been downloaded
+    """
+
+    filepath: Path
+    query: dict
+    request_id: str = None  # type: ignore # TODO: fix
+    downloaded: bool = False
+
+
+class Glofas(DataSource):
```

Comment:
If the children classes were in the same module (i.e. file) then yes, otherwise I don't think you're supposed to access protected classes outside of their own module.